src/utils/hooks/api/loans/get-usd-values.ts

```diff
@@ -1,35 +1,29 @@
-import { LendPosition, LoanPosition } from '@interlay/interbtc-api';
+import { BorrowPosition, LendPosition } from '@interlay/interbtc-api';
 import Big from 'big.js';
 
 import { convertMonetaryAmountToValueInUSD } from '@/common/utils/utils';
 import { getTokenPrice } from '@/utils/helpers/prices';
 
 import { Prices } from '../use-get-prices';
 
-const getTotalEarnedInterestUSDValue = (lendPositions: LendPosition[], prices: Prices): Big =>
-  lendPositions.reduce((totalValue: Big, position: LendPosition) => {
-    const { currency, earnedInterest } = position;
-    const price = getTokenPrice(prices, currency.ticker)?.usd;
+type Field<T> = T extends LendPosition
+  ? Extract<keyof LendPosition, 'earnedInterest' | 'earnedReward' | 'amount'>
+  : Extract<keyof BorrowPosition, 'earnedReward' | 'amount'>;
 
-    if (price === undefined) {
-      console.error(`useGetAccountCollateralization: No exchange rate found for currency: ${currency.name}`);
-    }
-
-    const positionUSDValue = convertMonetaryAmountToValueInUSD(earnedInterest, price);
-    return positionUSDValue === null ? totalValue : totalValue.add(positionUSDValue);
-  }, Big(0));
-
-const getTotalUSDValueOfPositions = (positions: LoanPosition[], prices: Prices): Big =>
-  positions.reduce((totalValue: Big, position: LoanPosition) => {
-    const { currency, amount } = position;
-    const price = getTokenPrice(prices, currency.ticker)?.usd;
+const getPositionsTotalUSDField = <T extends LendPosition | BorrowPosition>(
```

Comment:
This name seems little funky :) can you change it to something like `getSumOfFieldsInUSD` or similar?

Reply:
Can you also add comment to explain what this function is doing for future reference please?